### PR TITLE
Closes #3096 Simple file system configuration change to sanitize uploaded filenames.

### LINKED ISSUE #3096: Configure filename sanitization
## Motivation
Filename tranliteration is now in core as of Drupal 10.2

### Is your feature request related to a problem? Please describe.
Files with names that contain special characters, spaces and case sensitivity have long been a problem plaguing websites.

## Proposed Resolution
Configure the new settings 
image

### Describe the solution you'd like
image

### Describe alternatives you've considered
Let site owners configure this on their own.

### PR DESCRIPTION
**Pull request by maine-inventor**
## Related issues

#3096 

## How to test

Add any node content that contains a media file field.
Select file upload and select a local image file with a whacky file name ( spaces, special characters)
[upload]
Observe the new file name all cleaned up. Example: spaces are now "-"s. 
Also test the replacing of document files while preserving the filename (using the functionality from the Media Entity File Replace module).
Updated: Uploaded file character casing remains untouched.

### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
   - [ ] Bug fix
   - [x] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
   - [ ] Adding experimental module
   - [ ] Update experimental module
- **Minor release changes**
   - [ ] New feature
   - [x] Breaking or visual change to existing behavior
   - [ ] Upgrade experimental module to stable
   - [ ] Enable existing module by default or database update
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- **Other or unknown**
   - [ ] Other or unknown

### Drupal core
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major or minor level update
- **Other or unknown**
   - [ ] Other or unknown

### Drupal contrib projects
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major level update
- **Other or unknown**
   - [ ] Other or unknown

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: config/install/file.settings.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+_core:
+  default_config_hash: 2bwjZB1IjEYbppgZT3g7YW_5h_LDZYNa3DaDEfWX82U
```
Comment (bberndt-uaz) [inline config/install/file.settings.yml:2]: These lines can be deleted:
```suggestion
_core:
  default_config_hash: 2bwjZB1IjEYbppgZT3g7YW_5h_LDZYNa3DaDEfWX82U
```

Comment (maine-inventor) [reply config/install/file.settings.yml:2]: Thanks Brian, will update.

File: config/install/file.settings.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+icon:
+  directory: core/modules/file/icons
+make_unused_managed_files_temporary: false
+filename_sanitization:
+  transliterate: true
+  replace_whitespace: true
+  replace_non_alphanumeric: true
+  deduplicate_separators: true
+  lowercase: true
+  replacement_character: '-'
```
[the remark this answers is not in the document]
Comment (trackleft) [reply config/install/file.settings.yml:13]: Add empty new line at end of file.

**Comment by bberndt-uaz** [comment]
Do we want to enable all the "Sanitize filenames" settings? I'm wondering in particular about the "Convert to lowercase" setting.

**Comment by trackleft** [comment]
Please commit the suggestion here: https://github.com/az-digital/az_quickstart/pull/3180#discussion_r1492850410
image

Please convert this PR from a draft 
image

To ready for review
image

**Comment by joeparsons** [comment]
> Do we want to enable all the "Sanitize filenames" settings? I'm wondering in particular about the "Convert to lowercase" setting.

I do think it would be good to enable these settings by default on new site installs.  We don't necessarily want to enable these settings on existing sites though...

**Comment by joeparsons** [comment]
I was thinking we did this in Quickstart 1 but it looks like all we did was include the [contrib Transliteration module](https://www.drupal.org/project/transliteration) but it wasn't installed by default.  When that module was installed on Quickstart 1 its default settings basically match what is in this PR.

**Review by trackleft — APPROVED** [review]

**Review by danahertzberg — APPROVED** [review]

**Comment by danahertzberg** [comment]
Check if this needs to be in the Quickstart folder. Test against existing site to ensure compatibility.

**Review by trackleft — APPROVED** [review]
Tested this, and it appears to apply the configuration update on sites that existed before this setting existed.

Site that I tested on began its life in drupal core 10.1.7
The setting edited in this PR was introduced in Drupal core 10.2.x

**Review by tadean — APPROVED** [review]

File: config/install/file.settings.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  length: 128
+icon:
+  directory: core/modules/file/icons
+make_unused_managed_files_temporary: false
+filename_sanitization:
+  transliterate: true
+  replace_whitespace: true
+  replace_non_alphanumeric: true
+  deduplicate_separators: true
+  lowercase: true
```
Comment (trackleft) [inline config/install/file.settings.yml:12]:
```suggestion
  lowercase: false
```

**Review by trackleft — DISMISSED** [review]

**Comment by maine-inventor** [comment]
Updated the character casing check to not force to lower.

**Review by danahertzberg — APPROVED** [review]

**Review by joeparsons — APPROVED** [review]

**Review by tadean — APPROVED** [review]

**Review by trackleft — APPROVED** [review]

**Review by bberndt-uaz — APPROVED** [review]

**Pull request merged**